### PR TITLE
Add repository webhook APIs

### LINKED ISSUE #2453: Support "add webhook to repository" via REST API
### Before submitting an issue to GitBucket I have first:
- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] searched for similar already existing issue
- [x] read the documentation and [wiki](https://github.com/gitbucket/gitbucket/wiki)

## Issue
**Impacted version**: 4.33.0

**Deployment mode**: From gitbucket docker image

For CI workflows and other automated use-cases, we need the ability to add a webhook to a repository through the REST API.  Webhook configuration does not appear to be supported over REST API yet.  

For what it's worth, it looks like the class that gives the functionality to the UI is here:
https://github.com/gitbucket/gitbucket/blob/master/src/main/scala/gitbucket/core/controller/RepositorySettingsController.scala#L234

### PR DESCRIPTION
**Pull request by onukura**
### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [ ] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [x] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

## About this PR

This PR add Webhook APIs below to resolve #2453.

- [List repository webhooks](https://docs.github.com/en/rest/reference/repos#list-repository-webhooks)
- [Create a repository webhook](https://docs.github.com/en/rest/reference/repos#create-a-repository-webhook)
- [Get a repository webhook](https://docs.github.com/en/rest/reference/repos#get-a-repository-webhook)
- [Update a repository webhook](https://docs.github.com/en/rest/reference/repos#update-a-repository-webhook)
- [Delete a repository webhook](https://docs.github.com/en/rest/reference/repos#delete-a-repository-webhook)

`HOOK_ID` column added to `WEB_HOOK` table.

### CODE REVIEW

**Comment by onukura** [comment]
hmm... Test for mysql migration has problems.
```
java.sql.SQLSyntaxErrorException: (conn=3) Incorrect table definition; there can be only one auto column and it must be defined as a key
```

File: src/main/resources/update/gitbucket-core_4.35.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<changeSet>
+  <addColumn tableName="WEB_HOOK">
+    <column name="HOOK_ID" type="int" nullable="false" autoIncrement="true" unique="true"/>
+  </addColumn>
+  <dropForeignKeyConstraint constraintName="IDX_WEB_HOOK_EVENT_FK0" baseTableName="WEB_HOOK_EVENT"/>
+  <dropPrimaryKey tableName="WEB_HOOK" constraintName="IDX_WEB_HOOK_PK"/>
```
Comment (takezoe) [inline src/main/resources/update/gitbucket-core_4.35.xml:7]: Need to drop `IDX_WEB_HOOK_PK` before adding `HOOK_ID` column?

Comment (onukura) [reply src/main/resources/update/gitbucket-core_4.35.xml:5]: @takezoe Thanks for your advice! Finally test passed...
Needed unique constraint for ex. primary key (3 columns) to make foreign key for Postgres.

Comment (takezoe) [reply src/main/resources/update/gitbucket-core_4.35.xml:5]: Oh, I see. Thansk for dealing with it.

**Comment by onukura** [comment]
wiki updated

**Comment by solvingj** [comment]
thank you so much for your work on this!

File: src/main/scala/gitbucket/core/api/CreateARepositoryWebhook.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  secret: Option[String]
+)
+
+/**
+ * https://docs.github.com/en/rest/reference/repos#create-a-repository-webhook
+ */
+case class CreateARepositoryWebhook(
+  name: String = "web",
+  config: CreateARepositoryWebhookConfig,
+  events: List[String] = List("push"),
```
Comment (solvingj) [inline src/main/scala/gitbucket/core/api/CreateARepositoryWebhook.scala:16]: Is this hard coded to push?  

In the GUI, the following are supported:

Create - Branch, or tag created.
Gollum - Wiki page updated.
Issue comment - Issue commented on.
Issues - Issue opened, closed.
Pull request - Pull request opened, closed, or synchronized.
Pull request review comment - Pull request diff commented on.
Push - Git push to a repository.

Comment (takezoe) [reply src/main/scala/gitbucket/core/api/CreateARepositoryWebhook.scala:16]: This is the default value of the case class. You can specify other values when you call API to create a webhook.

Comment (solvingj) [reply src/main/scala/gitbucket/core/api/CreateARepositoryWebhook.scala:16]: Ok, thanks, I tried that and it didn't work so I opened ticket:  https://github.com/gitbucket/gitbucket/issues/2596

Comment (onukura) [reply src/main/scala/gitbucket/core/api/CreateARepositoryWebhook.scala:16]: @solvingj please see https://github.com/gitbucket/gitbucket/issues/2596#issuecomment-726113271

**Pull request merged**